conans/client/command.py

```diff
@@ -124,7 +124,57 @@ def new(self, *args):
                         upload_url=args.ci_upload_url,
                         gitlab_gcc_versions=args.ci_gitlab_gcc,
                         gitlab_clang_versions=args.ci_gitlab_clang)
+                        
 
+    def pack(self, *args):
+        """ Packs a binary package.
+        """
+        parser = argparse.ArgumentParser(description=self.pack.__doc__, prog="conan pack")
+        parser.add_argument("reference", help='user/channel, or a full package reference'
+                                              ' (Pkg/version@user/channel), if name and version '
+                                              ' are not declared in the recipe')
+
+        parser.add_argument("--package_id", default=None, required=True, help="specify package id to pack.")
+
+        parser.add_argument("--out", default=os.getcwd(), help='Directory to output the packed binaries.')
+
+        parser.add_argument("--packname", default=None, help='Name used for packing. '
+                                                             'If ommited it will be automatically generated.')
+
+        # TODO: obtain the list of supported methods from the packing class
+        parser.add_argument("--target", default='zip',
+                            help='Scheme to use for packing the binary package.')
+
+        args = parser.parse_args(*args)
+
+        try:
+            reference = ConanFileReference.loads(args.reference)
+        except ConanException:
+            print("Could not load reference: %s" % str(args.reference))
+        else:
+            if reference:
+                ret = self._conan.search_packages(reference)
+                ordered_packages, reference, recipe_hash, package_query = ret
+
+                if args.package_id in ordered_packages:
+                    name, version, user, channel = get_reference_fields(args.reference)
+
+                    properties = ordered_packages[args.package_id]
+                    return self._conan.pack(reference,
+                                            name=name,
+                                            version=version,
+                                            user=user,
+                                            channel=channel,
+                                            package_id=args.package_id,
```

Comment:
You already have `name`, `version`, `user` and `channel` in `reference` which you pass. Why not extract them as needed in `pack()` and not pass so many arguments?

See `upload` command.

Reply:
It is there because initially a reference was not used, rather a bunch of parameters.